sbomtool/internal/buildroot/scanner.go

```
@@ -0,0 +1,182 @@
+		"duration_ms", result.ScanDuration.Milliseconds())
+
+	return result, nil
+}
+
+// GetNormalizedPaths extracts normalized paths from scan results for Syft file coordinate matching.
+func (sr *ScanResult) GetNormalizedPaths() []string {
+	var normalizedPaths []string
+	for _, normalized := range sr.NormalizedPaths {
+		normalizedPaths = append(normalizedPaths, normalized)
+	}
+	return normalizedPaths
+}
+
+// scanRecursive performs recursive directory scanning with depth control.
+func (s *Scanner) scanRecursive(currentPath string, depth int, result *ScanResult) error {
+	if depth > s.maxDepth {
+		return nil
+	}
+
+	entries, err := os.ReadDir(currentPath)
+	if err != nil {
+		slog.Warn("Failed to read directory", "path", currentPath, "error", err)
+		return nil
+	}
+
+	for _, entry := range entries {
+		fullPath := filepath.Join(currentPath, entry.Name())
+
+		if s.shouldExclude(fullPath) {
+			continue
+		}
+
+		info, err := entry.Info()
+		if err != nil {
+			slog.Warn("Failed to get file info", "path", fullPath, "error", err)
+			continue
+		}
+
+		result.TotalSize += info.Size()
+		result.AllFiles = append(result.AllFiles, fullPath)
+
+		switch {
+		case info.Mode().IsRegular():
+			result.RegularFiles = append(result.RegularFiles, fullPath)
+
+		case info.Mode()&os.ModeSymlink != 0:
+			result.SymbolicLinks = append(result.SymbolicLinks, fullPath)
+
+		case info.IsDir():
+			result.Directories = append(result.Directories, fullPath)
+			if err := s.scanRecursive(fullPath, depth+1, result); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+// shouldExclude checks if a path should be excluded from scanning based on configured patterns.
+func (s *Scanner) shouldExclude(path string) bool {
+	for _, pattern := range s.excludePatterns {
+		if matched, _ := filepath.Match(pattern, path); matched {
+			return true
+		}
+		if matched, _ := filepath.Match(pattern, filepath.Base(path)); matched {
+			return true
+		}
```

In here you are silently ignoring a potential error, if the pattern (which is a configurable input) isn't a valid pattern. You should either validate the patterns in `NewScanner` or handle the error here.